### ISSUE TITLE
target and id are required

### ISSUE THREAD

**Issue by Dreamsorcerer** [marmelab/react-admin#8704]
I'm really struggling to use ReferenceManyField. When I use it, the spinner just spins for several seconds and gives the 'target and id are required' error.

I can see with a breakpoint that id is undefined, but I can't tell where this id is meant to come from or what I can do to affect its value. I can't see anything in the documentation mentioning it.

```
Error: target and id are required
    m useGetManyReference.ts:90
    fetchFn query.js:300
    c retryer.js:95
    s retryer.js:149
    promise callback*6350/s/< retryer.js:145
    promise callback*c retryer.js:116
    s retryer.js:149
    promise callback*6350/s/< retryer.js:145
    promise callback*c retryer.js:116
    s retryer.js:149
    promise callback*6350/s/< retryer.js:145
    promise callback*c retryer.js:116
    s retryer.js:156
    fetch query.js:332
    executeFetch queryObserver.js:199
    onSubscribe queryObserver.js:40
    subscribe subscribable.js:16
    R useBaseQuery.js:60
    React 7
    O scheduler.production.min.js:13
    R scheduler.production.min.js:14
    6813 scheduler.production.min.js:14
    Webpack 12
query.js:358:22
```

**Comment by fzaninotto** [comment]
Hi, and thanks for your question. As explained in [the react-admin contributing guide](https://github.com/marmelab/react-admin/blob/master/.github/CONTRIBUTING.md), we use the GitHub issues for bugs and feature requests only. 

For support question ("How To", usage advice, or troubleshooting your own code), you have three options:

- **Community support**: Ask your question on [StackOverFlow](https://stackoverflow.com/questions/tagged/react-admin). This makes your question easy to find by the developer community. StackOverFlow has great SEO, gamification, voting, and reputation system. 
- **Community support**: Ask your question on [Discord](https://discord.gg/GeZF9sqh3N). Some react-admin users and contributors might have time to answer your questions.
- **Professional support**: If you're a [React-Admin Enterprise Edition](https://marmelab.com/ra-enterprise/#support) subscriber, you have unlimited access to the Professional Support service. This service guarantees that you'll get an answer shortly. If you're not a subscriber, please consider it! It's a good way to contribute to the sustainability of react-admin. 

So I'm closing this issue, and inviting you to ask your question in one of the support channels above.

**Comment by Dreamsorcerer** [comment]
Sure, but the bug report is also labelled as ' Something isn't working as expected. Please tell us!'. It is not working as expected. :P

**Comment by Dreamsorcerer** [comment]
https://stackoverflow.com/questions/75636981/target-and-id-are-required-error-with-referencemanyfield

I'm really struggling to see how this can just be an issue with me though. If it's not a bug in the code, then surely there is something missing from the documentation..

**Comment by Dreamsorcerer** [comment]
Solution is to add a `source` prop which should be the field that is referenced by the foreign key.

This does not appear anywhere in the ReferenceManyField documentation page and is not included in any of the examples. It looks to me like none of the examples should work and would produce the same error I received...

I think this should be reopened to get the documentation fixed.

**Comment by slax57** [comment]
@Dreamsorcerer The `source` prop is documented in the [props section of the ReferenceManyField documentation](https://marmelab.com/react-admin/ReferenceManyField.html#props).

As you can see, it has a default value of `'id'` which explains why we don't need it in most of our examples.

The fact that you had to explicitely add it to you code probably comes from something specific to your implementation, but in most cases you should be able to omit it.

**Comment by Dreamsorcerer** [comment]
> @Dreamsorcerer The `source` prop is documented in the [props section of the ReferenceManyField documentation](https://marmelab.com/react-admin/ReferenceManyField.html#props).

Right, sorry, I must have been looking at the nav menu only. It doesn't have it's own section. Though I'd still suggest rewording it to something like "Source field used for mapping to reference (useful when a foreign key refers to something other than the records id field)." The current description says it's the property to display, which is clearly wrong, it doesn't display anything from the property.

**Comment by slax57** [comment]
Yes you are right, the docs could be better at explaining this prop. I'll reopen this issue as a documentation issue then.
